### PR TITLE
icon smooth ss

### PR DESCRIPTION
**Pull request by xxalpha**
This prevents the server from hanging if there's too much smoothing going on at server start, as is the case with lavaland.

Fixes https://github.com/tgstation/-tg-station/issues/16051

### CODE REVIEW

**Comment by optimumtact** [comment]
Is it guaranteed to finish before roundstart or is it just a best effort thing?

File: code/__HELPERS/icon_smoothing.dm [hunk 1]
```diff
@@ -111,8 +111,10 @@
 /proc/smooth_icon(atom/A)
 	if(qdeleted(A))
 		return
-	spawn(0) //don't remove this, otherwise smoothing breaks
-		if(A && (A.smooth & SMOOTH_TRUE) || (A.smooth & SMOOTH_MORE))
+	if(!A || !A.smooth)
+		return
+	spawn(0)
```
Comment (optimumtact) [inline code/__HELPERS/icon_smoothing.dm]: maybe keep that comment?

Comment (xxalpha) [inline code/__HELPERS/icon_smoothing.dm]: It's not going to break if you remove it, it'll just hang whatever is calling it.

**Comment by xxalpha** [comment]
It doesn't smooth before roundstart, only when the subsystem is active (after round starts).

File: code/__HELPERS/icon_smoothing.dm (hunk 1 above)
Comment (RemieRichards) [inline code/__HELPERS/icon_smoothing.dm]: what about using waitfor = 0 instead?

Comment (xxalpha) [inline code/__HELPERS/icon_smoothing.dm]: Same performance, but if I use spawn() I can avoid it if the if checks are true.

**Comment by duncathan** [comment]
It might be a good idea to make it smooth z1 and z2 before roundstart, and then get the rest after.

**Comment by xxalpha** [comment]
Even with lavaland loaded, the subsystem only takes a second to go through the entire list with thousands of atoms. I can smooth z1 and z2 on initialize, but it won't make a big difference.

**Comment by duncathan** [comment]
Probably best to do it on initialize if only to account for differences in server hardware downstream.

**Pull request merged**